cpp/include/raft/core/execution_stream.hpp

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
```

Comment:
As mentioned above, please use `raft::resources` or consolide this into `raft/core/resource` APIs so that we can further facade away any and all resources.